x-pack/plugins/cases/server/services/user_actions/operations/create.ts

```diff
@@ -154,6 +162,42 @@ export class UserActionPersister {
     return this.buildAddDeleteUserActions(params, createPayload, UserActionTypes.tags);
   }
 
+  private buildCustomFieldsUserActions(params: TypedUserActionDiffedItems<CaseCustomField>) {
+    const createPayload: CreatePayloadFunction<
+      CaseCustomField,
+      typeof UserActionTypes.customFields
+    > = (items: CaseCustomFields) => ({ customFields: items });
+
+    const { originalValue: originalCustomFields, newValue: newCustomFields } = params;
+
+    const originalCustomFieldsKeys = new Set(
+      originalCustomFields.map((customField) => customField.key)
+    );
+
+    const compareValues = arraysDifference(originalCustomFields, newCustomFields);
+
+    const updatedCustomFieldsUsersActions = compareValues?.addedItems
+      .filter((customField) => {
```

Comment:
We need to filter out custom fields that were added by the UI as `null` for the first time but the user changed another field.